docs/install.rst

```diff
@@ -257,32 +257,60 @@ Virtual environment
 Another method of installing Streamlink in a non-system-wide way is
 using `virtualenv`_, which creates a user owned Python environment instead.
 
-.. code-block:: bash
+- Install with ``virtualenv`` and ``pip`` commands
+
+  .. code-block:: bash
+
+      # Create a new environment
+      virtualenv ~/myenv
+
+      # Activate the environment
+      source ~/myenv/bin/activate
+
+      # *Either* install the latest Streamlink release from PyPI in the virtual environment
+      pip install --upgrade streamlink
+
+      # *Or*, install the most up-to-date development version from master on GitHub
+      pip install --upgrade git+https://github.com/streamlink/streamlink.git
+
+      # Use Streamlink in the environment
+      streamlink ...
+
+      # Deactivate the environment
+      deactivate
+
+      # Use Streamlink without activating the environment
+      ~/myenv/bin/streamlink ...
+
+- Install with ``pipx`` command
 
-    # Create a new environment
-    virtualenv ~/myenv
+  The `pipx`_ command combines the functionality of the ``virtualenv`` and ``pip`` commands. It may be necessary to
+  install it first, either with a system package manager, or using ``pip``, as detailed in the `documentation <pipx_>`_.
 
-    # Activate the environment
-    source ~/myenv/bin/activate
+  By default, ``pipx`` will install packages into ``~/.local/pipx/venvs/PACKAGE_NAME`` and create symbolic links in
+  ``~/.local/bin``. If these directories do not exist, they will be created. If ``~/.local/bin`` is newly created,
+  logging out and logging back in again may update the ``$PATH`` environment variable. Sourcing the relevant shell
+  configuration file in the current session may also update ``$PATH``. Otherwise, follow the directions in the note
+  output by ``pipx`` to update ``$PATH`` on your system.
```

Comment:
This is way too specific for Streamlink's install page. Linking to the pipx docs is more than enough:
https://pypa.github.io/pipx/docs/

I'm also not sure if this is what it's doing on Windows or macOS, too, as I've never used pipx in my Windows or macOS VMs.